deployment/cdk/app.py

```diff
@@ -254,31 +269,49 @@ def __init__(
                 "URS_PASSWORD": ecs.Secret.from_ssm_parameter(secret_urs_password),
             },
             environment=task_env,
-            logging=ecs.LogDrivers.aws_logs(stream_prefix=stack_id)
+            logging=ecs.LogDrivers.aws_logs(stream_prefix=stack_name)
         )
 
-        if settings.stage == "production":
-            hostname_part = ".ops"
-        else:
-            hostname_part = f".{settings.stage}"
+        load_balancer = elb.ApplicationLoadBalancer(self,
+            "mmt-lb",
+            vpc=vpc,
+            internet_facing=True
+        )
+
+        http_listener=load_balancer.add_listener('HttpListener',
+            protocol=elb.ApplicationProtocol.HTTP,
+            port=80
+        )
+        http_listener.add_redirect_response(
+            'HttpRedirect',
+            status_code='HTTP_301',
+            protocol='HTTPS',
+            port="443"
+        )
+
+        if settings.certificate_arn is not None and settings.certificate_arn != '':
+            certificate=certificatemanager.Certificate.from_certificate_arn(
+                self,
+                f"mmt-{settings.stage}-certificate",
+                settings.certificate_arn
+            )
 
         fargate_service = ecs_patterns.ApplicationLoadBalancedFargateService(
             self,
-            f"{stack_id}-service",
+            f"{stack_name}-service",
             cluster=cluster,
             desired_count=mincount,
-            public_load_balancer=True,
-            protocol=elb.ApplicationProtocol.HTTPS,
-            domain_name=f"mmt{hostname_part}.maap-project.org",
-            domain_zone=aws_route53.HostedZone.from_lookup(
-                self, f"{stack_id}-hosted-zone",
-                domain_name="maap-project.org"),
-            redirect_http=True,
-            task_definition=task_definition
+            task_definition=task_definition,
+            load_balancer=load_balancer,
+            listener_port=443,
+            certificate=certificate
```

Comment:
Using the `certificate` allows us to create HTTPS load balancers without having to try looking up the hosted zones in other AWS accounts, which isn't possible. 

See https://github.com/aws/aws-cdk/commit/e6c85e4167cdb38ed056eda17b869e179a6dd1c5